File: zig/bob/bob.zig
const std = @import("std");
const ascii = @import("std").ascii;

pub fn response(s: []const u8) []const u8 {
    const trim_s = std.mem.trim(u8, s, &ascii.whitespace);

    if (trim_s.len == 0)
        return "Fine. Be that way!";

    const is_question = trim_s[trim_s.len - 1] == '?';
    const is_capital = isAllCapital(s);

    if (is_question and is_capital)
        return "Calm down, I know what I'm doing!";

    if (is_question)
        return "Sure.";

    if (is_capital)
        return "Whoa, chill out!";

    return "Whatever.";
}

fn isAllCapital(s: []const u8) bool {
    var upper_flag = false;
    for (s) |c| {
        if (ascii.isLower(c)) return false;
        if (ascii.isUpper(c)) upper_flag = true;
    }
    return upper_flag;
}

File: python/bob/bob.py
def response(hey_bob: str) -> str:
    strip_hey_bob = hey_bob.strip()
    if not strip_hey_bob:
        return "Fine. Be that way!"
    if strip_hey_bob[-1] == '?':
        if strip_hey_bob.isupper():
            return "Calm down, I know what I'm doing!"
        return "Sure."
    if strip_hey_bob.isupper():
        return "Whoa, chill out!"
    return "Whatever."